docs/RELEASE.md

```diff
@@ -0,0 +1,46 @@
+# Release cycle and versioning
+
+## Versioning
+Versioning is using SemVer (X.Y.Z):
+- The X is a **major version** and is incremented when:
```

Comment:
From our comments: "This should only be incremented on breaking changes that are backwards incompatible" - I believe all of the items match that